sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -1215,10 +1216,19 @@ public StreamAppendClient getStreamAppendClient(String streamName, Descriptor de
         throws Exception {
       ProtoSchema protoSchema =
           ProtoSchema.newBuilder().setProtoDescriptor(descriptor.toProto()).build();
+      String dataflowJobId = GceMetadataUtil.fetchDataflowJobId();
```

Comment:
I wouldn't do this on every call to  get. Could  you instead embed an instance of  BigueryIOMetadata in DatasetServiceImpl, as is done in JobServiceImpl?

TBH, I  don't love relying on GCE metadata  here. Dataflow may decide in the future to  run on clusters other than  GCE, at which point this code will break. However as long as we keep things isolated.